Automattic/sensei
# Redirect to next lesson after completing a lesson

Addresses #1948.

To avoid the "Cannot modify header information - headers already sent" error after a postback, this PR switches from using the `wp_head` hook to using the `init` hook for `sensei_complete_lesson`. To facilitate that, the lesson ID is added to the POST request.

## Testing
1. Add some lessons to a course.
2. Register for the course and complete the first lesson.
3. After clicking the _Complete Lesson_ button, you should be automatically taken to the next lesson.
4. Add a quiz to one of the uncompleted lessons and ensure the _Allow user to retake the quiz_ checkbox is selected in the _Quiz Settings_.
5. Complete the quiz.
6. Navigate back to the lesson containing the quiz and click the _Reset Lesson_ button.
7. Ensure the lesson is reset and that the page remains on the current lesson.
8. Complete all of the lessons in the course. When clicking _Complete Lesson_ on the last lesson of the course, the page should remain on the current lesson.
